### ISSUE TITLE
print Initial orbital occupation?

### ISSUE THREAD

**Issue by hokru** [psi4/psi4#1711]
It can be very helpful to know the orbital occupations per IRREP before the SCF.
Otherwise one was has to wait until the end of the SCF or when it changes during the iterations.
I suggest to add a printout of the occupations after the SCF guess. 

This is a small change I think, just a print statement, and if this is wanted I can make the PR myself.

**Comment by dgasmith** [comment]
When we print out the original `nom` matrix, maybe we can show guess occupations there instead of 0's?

**Comment by hokru** [comment]
That would be a good location. Easy to move the printout to after the guess stage.

Though I didn't take into account that the `SAD` guess does not give orbitals and the occupations look strange. Not sure what it being printed.

**Comment by susilehtola** [comment]
> Though I didn't take into account that the `SAD` guess does not give orbitals and the occupations look strange. Not sure what it being printed.

Garbage? :smile:

**Comment by dgasmith** [comment]
With SAD, canonical operations are built and then munged. We store the canonical occupations in another temporary array... err forgot the name, should be at the bottom of `sad.cc`.

**Comment by hokru** [comment]
water in Cs symmetry:
SAD
```   -------------------------------------------------------
    Irrep   Nso     Nmo     Nalpha   Nbeta   Ndocc  Nsocc
   -------------------------------------------------------
     A'        28      28       6       6       6       0
     A"        11      11       1       1       1       0
   -------------------------------------------------------
    Total      39      39       7       7       7       0
   -------------------------------------------------------
```

SADNO (correct occ)
```   -------------------------------------------------------
    Irrep   Nso     Nmo     Nalpha   Nbeta   Ndocc  Nsocc
   -------------------------------------------------------
     A'        28      28       4       4       4       0
     A"        11      11       1       1       1       0
   -------------------------------------------------------
    Total      39      39       5       5       5       0
   -------------------------------------------------------
```

**Comment by dgasmith** [comment]
Right there are temporary arrays where you will have to change the default printing from `nalpha/nbeta` to `_nalpha_holder` or whatever for SAD.

**Comment by susilehtola** [comment]
For SAD you should only print out the occupations after the 1st iteration, since that's when it determines the real occupations.

**Comment by hokru** [comment]
SAD is annoying :-(

Maybe it is best to keep printing zeros for SAD (somehow) and additional a note saying that no occupations are available yet. For difficult open-shell systems, where this table becomes interesting, people might switch the guess anyway.

**Comment by susilehtola** [comment]
Yes, zeros for SAD would be an appropriate choice. I forget, does Psi4 print them out in the first iteration, like it does when it changes the occupations?

**Comment by hokru** [comment]
my PR suggests a smaller table for SAD, scraping the zero's altogether. Even less confusion.

> I forget, does Psi4 print them out in the first iteration, like it does when it changes the occupations?

No, it does not print them out.

**Comment by loriab** [comment]
closed by #1713
